source/extensions/transport_sockets/tls/ssl_socket.h

```diff
@@ -51,9 +52,10 @@ class SslExtendedSocketInfoImpl : public Envoy::Ssl::SslExtendedSocketInfo {
       Envoy::Ssl::ClientValidationStatus::NotValidated};
 };
 
-class SslSocketInfo : public Envoy::Ssl::ConnectionInfo {
+class SslSocketInfo : public Envoy::Ssl::ConnectionInfo, public Envoy::Ssl::Handshaker {
```

Comment:
Given what this class does, `SslHandshakerImpl` is probably a better name for this class?

Reply:
Done.